frontend/src/components/smart-contracts/SmartContractsProvider.tsx

```diff
@@ -20,25 +21,32 @@ export const SmartContractsProvider = ({
   }, [wallet]);
 
   // listen to ERC20 token approval status when the blockchain changes
-  const { isERC20Approved, setIsERC20Approved } = useERC20Approval(
-    wallet,
-    initialContracts
-  );
+  const {
+    isERC20Approved: isENGIApproved,
+    setIsERC20Approved: setIsENGIApproved,
+  } = useERC20Approval(wallet, initialContracts, SupportedTokens.ENGI);
+
+  const {
+    isERC20Approved: isUSDCApproved,
+    setIsERC20Approved: setIsUSDCApproved,
+  } = useERC20Approval(wallet, initialContracts, SupportedTokens.USDC);
```

Comment:
Both `useERC20Approval` methods are being called when either token is approved on the blockchain.  `useERC20ApprovalEventsFilter` in useENGIEvents.ts needs to be made token specific.